pkg/agent/multicast/mcast_discovery.go

```diff
@@ -138,6 +142,28 @@ func (s *IGMPSnooper) validate(event *mcastGroupEvent) (bool, error) {
 	return true, nil
 }
 
+func (s *IGMPSnooper) addToIGMPReportNPStatsMap(item types.McastNPValidationItem, packetLen uint64) {
+	uuid := item.Uuid
+	name := item.Name
+	ruleType := *item.NPType
+	if s.IGMPReportNPStats[ruleType][uuid] != nil {
```

Comment:
This is accessed by a single goroutine, and we need not locking?

Reply:
updated. Locking is needed.